### PR TITLE
Update tests, add Etherscan verification

### PR DESCRIPTION
**Pull request by grp06**
… Add etherscan verification

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -66,6 +68,11 @@ const config: HardhatUserConfig = {
       verbose: true,
     },
   },
+  etherscan: {
+    apiKey: {
+      goerli: ETHERSCAN_API_KEY!,
```
Comment (grp06) [inline hardhat.config.ts:73]: @rahulrumalla we need to add this to this repo's secrets in Github (and if it doesn't read the variable, figure out why)

Comment (rahulrumalla) [reply hardhat.config.ts:73]: Got it. Added now

**Review by rahulrumalla — APPROVED** [review]
LGTM

Although, does anything .openzeppelin file need to be added to gitignore?
https://docs.openzeppelin.com/cli/2.8/configuration#configuration-files-in-version-control

File: src/Badges.sol [hunk 2]
```diff
@@ -11,9 +9,7 @@ import "../lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgrad
 import "../lib/openzeppelin-contracts-upgradeable/contracts/utils/cryptography/draft-EIP712Upgradeable.sol";
 import "../lib/openzeppelin-contracts-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
 import "../lib/openzeppelin-contracts-upgradeable/contracts/utils/introspection/ERC165Upgradeable.sol";
-import { ERC165 } from "./ERC165.sol";
 import { IERC721Metadata } from "./IERC721Metadata.sol";
-
 bytes32 constant AGREEMENT_HASH = keccak256("Agreement(address active,address passive,string tokenURI)");
 
 contract Badges is
```
Comment (rahulrumalla) [inline src/Badges.sol:15]: Shouldn't this contract still be implementing the IERC4973 interface? Was this missed?

**Comment by rahulrumalla** [comment]
- [ ] I noticed that you're now committing changes to lib/ERC4973, did u intend to? 
- [ ] Please move all "v2" files into a tests/proxy folder to separate this from actual source code

**Review by rahulrumalla — APPROVED** [review]

**Pull request merged**